MonoGame.Framework/Utilities/Png/PngWriter.cs

```diff
@@ -211,6 +211,16 @@ private void GetColorData(Texture2D texture2D)
 
             switch (texture2D.Format)
             {
+                case SurfaceFormat.Single:
+                    var floatData = new float[colorDataLength];
+                    texture2D.GetData<float>(floatData);
+                    
+                    for (int i = 0; i < colorDataLength; i++) {
+                        // Export as a greyscale image.
+                        colorData[i] = new Color(floatData[i], floatData[i], floatData[i]);
```

Comment:
I've just performed a small array manipulation benchmark summing an array of 1000000 integers, repeating this test 1000 times. The results are as follows:

normal
mean: 4.314 ms
stddev: 1.078 ms

unsafe with fixed
mean: 4.421 ms
stddev: 0.9709 ms

As you can see, there is a slight performance benefit, though I'm not sure if it's enough to justify the readability decrease. Any thoughts?